google-tag-manager.html

```diff
@@ -22,41 +22,109 @@
 -->
 
 <dom-module id="google-tag-manager">
-  <template>
+    <template>
     <noscript>
       <iframe src={{_computeSrc(cid)}} height="0" width="0" style="display:none;visibility:hidden"></iframe>
     </noscript>
   </template>
 
-  <script>
-    Polymer({
-      is: 'google-tag-manager',
-
-      properties: {
-        /**
-         * `cid` is the container ID
-         */
-        cid: {
-          type: String,
-        }
-      },
-
-      // Element Lifecycle
-
-      attached: function() {
-        (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
-        new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
-        j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
-        '//www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
-        })(window,document,'script','dataLayer',this.cid);
-      },
-
-
-      // Element Behavior
-
-      _computeSrc: function(cid) {
-        return '//www.googletagmanager.com/ns.html?id=' + cid;
-      }
-    });
-  </script>
+    <script>
+        Polymer({
+            is: 'google-tag-manager',
+            timeout: false,
+            isAnalyticsLoaded: false,
+            offlineEvents: [],
+            properties: {
+                /**
+                 * `cid` is the container ID
+                 */
+                cid: {
+                    type: String,
+                }
+            },
+
+            // Element Lifecycle
+
+            attached: function() {
+                (function(w, d, s, l, i) {
+                    w[l] = w[l] || [];
+                    w[l].push({
+                        'gtm.start': new Date().getTime(),
+                        event: 'gtm.js'
+                    });
+                    var f = d.getElementsByTagName(s)[0],
+                        j = d.createElement(s),
+                        dl = l != 'dataLayer' ? '&l=' + l : '';
+                    j.async = true;
+                    j.src =
+                        '//www.googletagmanager.com/gtm.js?id=' + i + dl;
+                    f.parentNode.insertBefore(j, f);
+                })(window, document, 'script', 'dataLayer', this.cid);
+            },
+            _checkGA: function() {
+                if (typeof ga !== "function") {
+                    return;
+                }
+                clearInterval(this.interval);
+                this.isAnalyticsLoaded = true;
+                this._processOfflineEvents.bind(this)();
+            },
+
+            _processOfflineEvents: function() {
+                // is ga ok?
+                this.offlineEvents.map(function sendEvent(ev) {
+                    ga('send', 'event', ev.category, ev.event, ev.label);
+                });
+                //empty the array
+                this.offlineEvents.length = 0;
+            },
+
+            createLabel: function(category, event, sessionId, customerId, attributes) {
+
+                function guid() {
```

Comment:
this is the fn I told you, I find it's way more resilient, as it uses the browser signature as part of the uid.

https://github.com/ericelliott/cuid

> from the repo docs :P

Broken down

** c - h72gsb32 - 0000 - udoc - l363eofy **

The groups, in order, are:

'c' - identifies this as a cuid, and allows you to use it in html entity ids.
Timestamp
Counter - a single process might generate the same random string. The weaker the pseudo-random source, the higher the probability. That problem gets worse as processors get faster. The counter will roll over if the value gets too big.
Client fingerprint
Pseudo random (Math.random() in JavaScript)